Lib/ufo2ft/featureWriters/cursFeatureWriter.py

```diff
@@ -0,0 +1,129 @@
+        if any(unicodeScriptDirection(uv) == "LTR" for uv in cmap):
+            gsub = self.compileGSUB()
+            dirGlyphs = classifyGlyphs(unicodeScriptDirection, cmap, gsub)
+            shouldSplit = "LTR" in dirGlyphs
+        else:
+            shouldSplit = False
+
+        lookups = []
+        ordereredGlyphSet = self.getOrderedGlyphSet().items()
+        if shouldSplit:
+            # Make LTR lookup
+            LTRlookup = self._makeCursiveLookup(
+                (
+                    glyph
+                    for (glyphName, glyph) in ordereredGlyphSet
+                    if glyphName in dirGlyphs["LTR"]
+                ),
+                direction="LTR",
+            )
+            if LTRlookup:
+                lookups.append(LTRlookup)
+
+            # Make RTL lookup with other glyphs
+            RTLlookup = self._makeCursiveLookup(
+                (
+                    glyph
+                    for (glyphName, glyph) in ordereredGlyphSet
+                    if glyphName not in dirGlyphs["LTR"]
+                ),
+                direction="RTL",
+            )
+            if RTLlookup:
+                lookups.append(RTLlookup)
+        else:
+            lookup = self._makeCursiveLookup(
+                (glyph for (glyphName, glyph) in ordereredGlyphSet)
+            )
+            if lookup:
+                lookups.append(lookup)
+
+        if lookups:
+            feature = ast.FeatureBlock("curs")
+            feature.statements.extend(lookups)
+            return feature
+
+    def _makeCursiveLookup(self, glyphs, direction=None):
+        statements = self._makeCursiveStatements(glyphs)
+
+        if not statements:
+            return
+
+        suffix = ""
+        if direction == "LTR":
+            suffix = "_ltr"
+        elif direction == "RTL":
+            suffix = "_rtl"
+        lookup = ast.LookupBlock(name=f"curs{suffix}")
+
+        if direction != "LTR":
+            lookup.statements.append(
+                ast.makeLookupFlag(flags=("IgnoreMarks", "RightToLeft"))
+            )
+        else:
+            lookup.statements.append(ast.makeLookupFlag(name="IgnoreMarks"))
+
+        lookup.statements.extend(statements)
+
+        return lookup
+
+    def _makeCursiveStatements(self, glyphs):
+        cursiveAnchors = dict()
+        statements = []
+        for glyph in glyphs:
+            entryAnchor = exitAnchor = None
+            for anchor in glyph.anchors:
+                if entryAnchor and exitAnchor:
+                    break
+                if anchor.name in self.entryAnchorNames:
+                    entryAnchor = ast.Anchor(x=otRound(anchor.x), y=otRound(anchor.y))
+                elif anchor.name in self.exitAnchorNames:
+                    exitAnchor = ast.Anchor(x=otRound(anchor.x), y=otRound(anchor.y))
+
+            if entryAnchor or exitAnchor:
+                cursiveAnchors[ast.GlyphName(glyph.name)] = (entryAnchor, exitAnchor)
```

Comment:
so it is possible that one of the anchors is None, what's the meaning of that?

Reply:
oh ok, it means that glyph is only at the beginning or end, not middle

Reply:
Maybe that's worth a comment?